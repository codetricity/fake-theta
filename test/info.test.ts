import { request, toMatchInvalidHeaderParameterError } from './helper';

describe('GET /osc/info', () => {
  it('should respond with info', async () => {
    const res = await request().get('/osc/info').expect(200);
    expect(res.body).toMatchSnapshot();
  });
});

describe('THETA X GET /osc/info', () => {
  it('should respond with info', async () => {
    const res = await request()
      .get('/osc/info')
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });
});

describe('THETA Z1 GET /osc/info', () => {
  it('should respond with info', async () => {
    const res = await request()
      .get('/osc/info')
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body).toMatchSnapshot();
  });
});

describe('Unknown Type GET /osc/info', () => {
  it('should respond invalidHeaderParameterError', async () => {
    const res = await request()
      .get('/osc/info')
      .set('emulating-theta-model', 'unknown')
      .expect(400);
    toMatchInvalidHeaderParameterError(res);
  });
});
