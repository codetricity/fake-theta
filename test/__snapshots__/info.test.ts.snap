// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /osc/info should respond with info 1`] = `
{
  "_bluetoothMacAddress": "00:45:de:78:3e:33",
  "_wlanMacAddress": "00:45:78:bc:45:67",
  "api": [
    "/osc/info",
    "/osc/state",
    "/osc/checkForUpdates",
    "/osc/commands/execute",
    "/osc/commands/status",
  ],
  "apiLevel": [
    2,
  ],
  "endpoints": {
    "httpPort": 80,
    "httpUpdatesPort": 80,
  },
  "firmwareVersion": "1.41.0",
  "gps": false,
  "gyro": false,
  "manufacturer": "Ricoh Company, Ltd.",
  "model": "RICOH THETA X",
  "serialNumber": "10100001",
  "supportUrl": "https://theta360.com/en/support/",
  "uptime": 67,
}
`;

exports[`THETA X GET /osc/info should respond with info 1`] = `
{
  "_bluetoothMacAddress": "00:45:de:78:3e:33",
  "_wlanMacAddress": "00:45:78:bc:45:67",
  "api": [
    "/osc/info",
    "/osc/state",
    "/osc/checkForUpdates",
    "/osc/commands/execute",
    "/osc/commands/status",
  ],
  "apiLevel": [
    2,
  ],
  "endpoints": {
    "httpPort": 80,
    "httpUpdatesPort": 80,
  },
  "firmwareVersion": "1.41.0",
  "gps": false,
  "gyro": false,
  "manufacturer": "Ricoh Company, Ltd.",
  "model": "RICOH THETA X",
  "serialNumber": "10100001",
  "supportUrl": "https://theta360.com/en/support/",
  "uptime": 67,
}
`;

exports[`THETA Z1 GET /osc/info should respond with info 1`] = `
{
  "_bluetoothMacAddress": "00:45:de:78:3e:33",
  "_wlanMacAddress": "00:45:78:bc:45:67",
  "api": [
    "/osc/info",
    "/osc/state",
    "/osc/checkForUpdates",
    "/osc/commands/execute",
    "/osc/commands/status",
  ],
  "apiLevel": [
    2,
  ],
  "endpoints": {
    "httpPort": 80,
    "httpUpdatesPort": 80,
  },
  "firmwareVersion": "2.20.3",
  "gps": false,
  "gyro": true,
  "manufacturer": "RICOH",
  "model": "RICOH THETA Z1",
  "serialNumber": "10100001",
  "supportUrl": "https://theta360.com/en/support/",
  "uptime": 67,
}
`;
